Feature/twitter data

gqinami opened the pull request:
Two major functionalities: 
- Connect via socket to receive twitter data (i.e. Mobs)
- Implement filter function (currently only for sightings, since predictions are still not available)

oleks-fedotov committed:
trying to set up socket connection

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

Conflicts:
	.gitignore
	module/main.js
	package.json

oleks-fedotov committed:
playing around

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

Conflicts:
	module/main.js
	script.js

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

Conflicts:
	package.json

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

Conflicts:
	module/main.js

oleks-fedotov committed:
twitter data visualized

oleks-fedotov committed:
marker improved + better distribution

gqinami committed:
get past data by time range filter

gqinami committed:
fetch data function for getting pokemon data by time range

gqinami committed:
changes in main.js for filter by time

gqinami committed:
- changes in script.js to add filter options
- changes in the updatePoints function to show all pokemons if the array of pokemonIds is null

gqinami committed:
creates the filter function separately, so it can be called many times, after the map is initialized

gqinami committed:
change the leaflet version to 0.7.7

gqinami committed:
change the app version

johartl commented on module/main.js:
```diff
@@ -31,9 +33,13 @@ require('../style.css');
         var zoomLevel = options.zoomLevel;
         var timeRange = options.timeRange;
         var apiEndpoint = options.apiEndpoint;
+        var socketEndPoint = options.webSocketEndPoint;
```
The option is called `websocketEndpoint`

johartl commented on module/main.js:
```diff
@@ -242,25 +264,71 @@ require('../style.css');
 
         }
 
+        function filter(pokemonIds, sightingsSince, predictionsUntil) {
```
The filter function receives only one parameter which is an object containing the three filter criteria:

```
function filter(filterOptions) {
  let pokemonIds = filterOptions.pokemonIds;
  let sightingsSince = filterOptions.sightingsSince;
  let predictionsUntil = filterOptions.predictionsUntil;
  ...
}
```

johartl commented on script.js:
```diff
@@ -1,7 +1,11 @@
+        pokemonIds: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], sightingsSince: new Date(2016, 5, 20), predictionsUntil: 0
+    },
     apiEndpoint: 'http://pokedata.c4e3f8c7.svc.dockerapp.io:65014',
+    webSocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
```
Again `websocketEndpoint`

gqinami committed:
changes as per Jochen's comments:
- pass an object as parameter to filter function;
- change the name of websocketEndpoint

johartl commented on module/main.js:
```diff
@@ -33,13 +33,10 @@ var DataService = require('./DataService.js');
         var timeRange = options.timeRange;
         var apiEndpoint = options.apiEndpoint;
-        var socketEndPoint = options.webSocketEndPoint;
+        var socketEndPoint = options.websocketEndPoint;
```
Sorry but it's `websocketEndpoint` (no capital P)

johartl commented on script.js:
```diff
@@ -2,10 +2,10 @@ var htmlElement = 'mapid';
     },
     apiEndpoint: 'http://pokedata.c4e3f8c7.svc.dockerapp.io:65014',
-    webSocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
+    websocketEndPoint: 'pokedata.c4e3f8c7.svc.dockerapp.io:65024',
```
Same here

johartl commented on module/main.js:
```diff
@@ -283,7 +280,11 @@ var DataService = require('./DataService.js');
+
+            var sightingsSince = options.filter.sightingsSince;
+            var predictionsUntil = options.filter.predictionsUntil;
+            var pokemonIds = options.filter.pokemonIds;
```
I think this should be

```
var sightingsSince = filterOptions.sightingsSince;
var predictionsUntil = filterOptions.predictionsUntil;
var pokemonIds = filterOptions.pokemonIds;
```

gqinami replied:
yeah you are right. Thanks for the comments

gqinami committed:
small changes to names of parameters in filter function

johartl commented:
Seems fine now

oleks-fedotov committed:
Merge branch 'develop' into feature/twitter-data

Conflicts:
	module/main.js

gqinami committed:
fix filter function to call filterbytime in case pokemonIds is null

gqinami committed:
Merge branch 'feature/twitter-data' of https://github.com/PokemonGoers/PokeMap-1 into feature/twitter-data
